salt/modules/etcd_mod.py

```diff
@@ -54,7 +62,7 @@ def __virtual__():
     """
     Only return if python-etcd is installed
     """
-    if salt.utils.etcd_util.HAS_LIBS:
+    if salt.utils.etcd_util.HAS_ETCD_V2 or salt.utils.etcd_util.HAS_ETCD_V3:
```

Comment:
Good suggestion, wouldn't be sure how to implement it though.  This config can be in either one of two places, the global config space, or under a profile (which they can have many of, each with their own setting for the `etcd.require_v2`, and is supplied for each call to `etcd_mod` if applicable).  So, in order to keep these as static globals for the module, I think it might have to stay like this (unless you have an idea here)

Reply:
I forget precisely the module - I remember something about profile and then getting config based on the profile if it was set.

Hrm. Seems like something we should have a util for :suspect:

Reply:
Okay, well I'm not seeing the example that I thought existed. I'll try checking tomorrow(?)